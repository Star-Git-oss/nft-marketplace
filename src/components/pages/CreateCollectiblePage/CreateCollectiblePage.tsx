import React from 'react';
import { useTranslation } from 'react-i18next';
import { useHistory } from 'react-router-dom';
import styled from 'styled-components';
import { Helmet } from 'react-helmet';
import { H1, H3, P } from '../../common/Title/Title';
import Col from '../../common/ui-library/Col/Col';
import Row from '../../common/ui-library/Row/Row';
import Card from '../../common/ui-library/Card/Card';
import TinyContainer from '../../common/ui-library/TinyContainer/TinyContainer';
import Colors from '../../common/ui-library/styles/colors';
import { GoBack } from '../../common/Utils/Utils';

const CategoryCard = styled(Card)`
  text-align: center;
  margin: 0 auto;
  & h3,
  div {
    text-align: center;
    margin: 0 auto;
    padding: 0;
  }
  &:hover {
    background: ${Colors.background_gray};
    cursor: pointer;
  }
`;

const SampleCard = styled.div`
  background: linear-gradient(180deg, #f29fff 0%, #878cff 100%);
  border-radius: 4px;
  height: 112px;
  width: 79px;
`;

const CreateCollectiblePage: React.FC = () => {

  const history = useHistory();

  const { t } = useTranslation();

  const createSingle = () => {
    history.push('/create-single-collectible');
  };  

  const createMultiple = () => {
    history.push('/create-multiple-collectible');
  };  

  return (
    <TinyContainer>

      <Helmet>
        <title>{t('createCollectible.seo.title')}</title>
        <meta name="description" content={t('createCollectible.seo.description')} />
        <meta name="keywords" content={t('createCollectible.seo.keywords')} />
      </Helmet>

      <GoBack text={t('createCollectible.goBack')} history={history} />

      <H1>{t('createCollectible.title')}</H1>

      <P>
        {t('createCollectible.introduction')}
      </P>

      <Row>
        <Col size="50">
          <CategoryCard onClick={createSingle}>
            <SampleCard />
            <H3>{t('createCollectible.single')}</H3>
          </CategoryCard>
        </Col>

        <Col size="50">
          <CategoryCard onClick={createMultiple}>
            <SampleCard />
            <H3>{t('createCollectible.multiple')}</H3>
          </CategoryCard>
        </Col>
      </Row>

      <P>
        {t('createCollectible.closing')}
      </P>
    </TinyContainer>
  );
};

export default CreateCollectiblePage;