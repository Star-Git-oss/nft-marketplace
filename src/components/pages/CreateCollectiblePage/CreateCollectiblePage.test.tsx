/* eslint-disable no-console */
import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import CreateCollectiblePage from './CreateCollectiblePage';

// Known issue: error and warning does not work for console
// https://github.com/facebook/react/issues/7047
const original = console.error;

beforeEach(() => {
  console.error = jest.fn();
});

afterEach(() => {
  console.error = original;
});

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key) => key }),
}));

const mockHistoryPush = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));


describe('CreateCollectiblePage snapshots', () => {

  test('CreateCollectiblePage renders correctly', () => {
    const tree = renderer
      .create(<CreateCollectiblePage />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

});

describe('CreateCollectiblePage', () => {

  it('renders component properly', () => {
    render(<CreateCollectiblePage />);

    // Validate first button for creatingle single element
    const single = screen.getByText('createCollectible.single');
    expect(single).toBeInTheDocument();
    fireEvent.click(single);
    expect(mockHistoryPush).toHaveBeenCalledWith('/create-single-collectible');

    // Validate second button for creatingle multiple elements
    const multiple = screen.getByText('createCollectible.multiple');
    expect(multiple).toBeInTheDocument();
    fireEvent.click(multiple);
    expect(mockHistoryPush).toHaveBeenCalledWith('/create-multiple-collectible');
  
  });

});