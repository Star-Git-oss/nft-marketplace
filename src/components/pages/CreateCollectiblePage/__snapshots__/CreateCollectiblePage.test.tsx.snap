// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateCollectiblePage snapshots CreateCollectiblePage renders correctly 1`] = `
<div
  className="sc-crrsfI gDQLBn"
>
  <div
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
      />
    </svg>
     Â 
    <span
      onClick={[Function]}
      role="button"
      style={
        Object {
          "verticalAlign": "top",
        }
      }
      tabIndex={0}
    >
      createCollectible.goBack
    </span>
  </div>
  <h1
    className="sc-dlfnbm jnRgnW"
  >
    createCollectible.title
  </h1>
  <p
    className="sc-iBPRYJ kEpCMi"
  >
    createCollectible.introduction
  </p>
  <div
    className="sc-kEjbxe jPZapw"
  >
    <div
      className="sc-jrAGrp hwipwK"
      size="50"
    >
      <div
        className="sc-iqHYGH kQHiEq sc-dQppl dPDMLy"
        onClick={[Function]}
      >
        <div
          className="sc-bqyKva fJrUzg"
        />
        <h3
          className="sc-eCssSg dNNVqm"
        >
          createCollectible.single
        </h3>
      </div>
    </div>
    <div
      className="sc-jrAGrp hwipwK"
      size="50"
    >
      <div
        className="sc-iqHYGH kQHiEq sc-dQppl dPDMLy"
        onClick={[Function]}
      >
        <div
          className="sc-bqyKva fJrUzg"
        />
        <h3
          className="sc-eCssSg dNNVqm"
        >
          createCollectible.multiple
        </h3>
      </div>
    </div>
  </div>
  <p
    className="sc-iBPRYJ kEpCMi"
  >
    createCollectible.closing
  </p>
</div>
`;
